tests_e2e/tests/extensions_disabled/extensions_disabled.py

```diff
@@ -47,27 +48,43 @@ def run(self):
+        # Set the timeout to the minimum allowed(15 minutes) so we can check that the extension fails before timeout
         log.info("Setting the extension timeout to 15 minutes")
         vm: VirtualMachineClient = VirtualMachineClient(self._context.vm)
-
         vm.update({"extensionsTimeBudget": "PT15M"})
 
         disabled_timestamp: datetime.datetime = datetime.datetime.utcnow() - datetime.timedelta(minutes=60)
 
         #
-        # Validate that the agent is not processing extensions by attempting to run CustomScript
+        # Validate that the agent is not processing extensions by attempting to run CSE & checking that provisioning fails fast
         #
-        log.info("Executing CustomScript; it should time out after 15 min or so.")
+        log.info("Executing CustomScript; the agent should report a VMExtensionProvisioningError without processing the extension")
         custom_script = VirtualMachineExtensionClient(self._context.vm, VmExtensionIds.CustomScript, resource_name="CustomScript")
         try:
-            custom_script.enable(settings={'commandToExecute': "date"}, force_update=True, timeout=20 * 60)
-            fail("CustomScript should have timed out")
+            # Use mkdir in script settings so we can check if the command was executed
+            custom_script.enable(settings={'commandToExecute': "mkdir /var/lib/waagent/testdir"}, force_update=True, timeout=6 * 60)
+            fail("The agent should have reported an error processing the goal state")
         except Exception as error:
-            assert_that("VMExtensionProvisioningTimeout" in str(error)) \
-                .described_as(f"Expected a VMExtensionProvisioningTimeout: {error}") \
+            assert_that("VMExtensionProvisioningError" in str(error)) \
+                .described_as(f"Expected a VMExtensionProvisioningError error, but actual error was: {error}") \
                 .is_true()
-            log.info("CustomScript timed out as expected")
+            assert_that("Extension will not be processed since extension processing is disabled" in str(error)) \
+                .described_as(f"Error message should communicate that extension will not be processed, but actual error "
+                              f"was: {error}").is_true()
+            log.info("Goal state processing failed as expected")
+
+        #
+        # Validate the agent is not processing extensions by checking it did not execute the extension settings
+        #
+        try:
+            ssh_client.run_command("dir /var/lib/waagent/testdir", use_sudo=True)
+            fail("The extension settings should not have been executed")
+        except CommandError as error:
+            assert_that(error.stderr)\
+                .described_as("Extension should not be processed, but extension settings were executed")\
+                .contains('No such file or directory')
+            log.info("The agent did not execute the extension settings as expected")
```

Comment:
how do you know that extensions were processed? CommandError just indicates a generic error (e.g. typo in the command, SSH timeout)

Reply:
Updated this to list the contents of /var/lib/waagent. If the testdir is there then we know the command in ext settings was executed. If it isn't then we know the extension wasn't processed.